src/rust/lib/enso-prelude/src/option.rs

```diff
@@ -3,9 +3,11 @@
 /// Adds mapping methods to the `Option` type.
 pub trait OptionOps {
     type Item;
-    fn map_ref      <U,F> (&self , f:F) -> Option<U> where F : FnOnce(&Self::Item) -> U;
-    fn for_each     <U,F> (self  , f:F)              where F : FnOnce(Self::Item)  -> U;
-    fn for_each_ref <U,F> (&self , f:F)              where F : FnOnce(&Self::Item) -> U;
+    fn map_ref      <U,F> (&self, f:F) -> Option<U> where F : FnOnce(&Self::Item) -> U;
+    fn for_each     <U,F> ( self, f:F)              where F : FnOnce(Self::Item)  -> U;
+    fn for_each_ref <U,F> (&self, f:F)              where F : FnOnce(&Self::Item) -> U;
+    /// Returns true if option contains Some with value matching given predicate.
+    fn contains_if  <F>   (&self, f:F) -> bool      where F : FnOnce(&Self::Item) -> bool;
```

Comment:
TBH I dont like this name - its very hard to tell what this function does without reading its body. I don't have a better name in my head though. Do we need to define it? Is it often needed? If not, maybe lets just use longer form for now?

Reply:
Personally I think `map_or(false, |x| ...)` have similar length, and does not introduce any new name to comprehend.

Reply:
@wdanilo yes, we do need it. @mwu-tow requested me to move this function to this top-level library because it needs to be used in other parts of the project instead of only using it in the `IDE` crate. I think `contains` would make more sense, but an experimental method already exists for it (https://doc.rust-lang.org/std/option/enum.Option.html#method.contains).

Reply:
Let's leave it for now, but I second @farmaazon . `map_or(false, |x| ...)` is the same length and imo easier to understand.